universum/modules/vcs/perforce_vcs.py

```diff
@@ -459,6 +459,13 @@ def p4unshelve(self, *args, **kwargs):
             raise
         return result
 
+    @catch_p4exception(ignore_if="No file(s) to resolve")
+    def p4resolve(self):
+        result = str(self.p4.run_resolve("-am"))
```

Comment:
In my opinion this operation and its outcome must be reported to both logs and repo status.